cypress-io/cypress
# fix: Encode fileServerFolder to fix #25839

- Closes https://github.com/cypress-io/cypress/issues/25839 

### Additional details
When there are invalid header characters in a test folder path (such as Chinese characters), setting the `x-cypress-file-path` header fails. This PR will encode the file path header value to solve the issue and decode it in `server-base.ts`.

### Steps to test
* Create a project that is under a folder that contains character not allowed in a header.
* Run a test that `cy.visit`s a file served from the local file server
* Verify the visit succeeds 

### How has the user experience changed?
The `Invalid character in header content ["x- cypress-file-path"]` error is no longer thrown.

### PR Tasks
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
